File: frontend/src/services/adminRegistrationService.js
import axiosInstance from '../config/axios';

export const adminRegistrationService = {
  getEventRegistrations: async (eventId) => {
    const response = await axiosInstance.get(`/api/admin/events/${eventId}/registrations`);
    return response.data;
  },

  getAllRegistrations: async () => {
    const response = await axiosInstance.get('/api/admin/events/registrations/all');
    return response.data;
  },

  manualCheckIn: async (registrationId) => {
    const response = await axiosInstance.post(`/api/admin/events/registrations/${registrationId}/check-in`);
    return response.data;
  },

  cancelCheckIn: async (registrationId) => {
    const response = await axiosInstance.post(`/api/admin/events/registrations/${registrationId}/cancel-check-in`);
    return response.data;
  },

  cancelRegistration: async (registrationId) => {
    const response = await axiosInstance.delete(`/api/admin/events/registrations/${registrationId}`);
    return response.data;
  }
};

File: backend/src/main/java/com/project/event/service/AdminRegistrationService.java
package com.project.event.service;

import com.project.event.dto.AdminRegistrationResponse;
import com.project.event.dto.MessageResponse;

import java.util.List;
import java.util.UUID;

public interface AdminRegistrationService {
    List<AdminRegistrationResponse> getEventRegistrations(UUID eventId);
    List<AdminRegistrationResponse> getAllRegistrations();
    MessageResponse manualCheckIn(UUID registrationId);
    MessageResponse cancelCheckIn(UUID registrationId);
    MessageResponse cancelStudentRegistration(UUID registrationId);
}
